### PR TITLE
Angular: add a couple `|null`s

### PR DESCRIPTION
**Pull request by ersimont**
Please fill in this template.

- [X] Make your PR against the `master` branch.
- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [ ] Run `tsc` without errors.
- [ ] Run `npm run lint package-name` if a `tslint.json` is present.

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes:
  For the `hash` method: https://docs.angularjs.org/api/ng/service/$location#hash. 
  For the `search` method: see the comment on that method that states the behavior when passing in `null`.
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

### CODE REVIEW

File: angular/index.d.ts [hunk 1]
```diff
@@ -901,8 +901,17 @@ declare namespace angular {
...
+        /**
+         * Returns the hash fragment
+         */
         hash(): string;
-        hash(newHash: string): ILocationService;
+
+        /**
+         * Changes the hash fragment and returns `$location`
+         */
+        hash(newHash: string|null): ILocationService;
```
Comment (sheetalkamat) [inline angular/index.d.ts]: Don't you want to rather make newHash as optional parameter?

Comment (thorn0) [inline angular/index.d.ts]: This function is a getter-setter. When invoked without parameters, it's a getter.

**Comment by thorn0** [comment]
:+1:

**Review by sheetalkamat — COMMENTED** [review]
@thorn0  and @ersimont If the function can be invoked without parameter, it should rather be changed from 
```typescript
hash(newHash: string|null): ILocationService;
```
to
```typescript
hash(newHash?: string): ILocationService;
```

**Comment by thorn0** [comment]
It can be invoked without parameters as a getter or with `null` as a setter.

**Comment by sheetalkamat** [comment]
If that's the case the definition needs to be 
```
hash(newHash?: string|null): ILocationService;
```
And same change for other method too.

**Comment by thorn0** [comment]
No. Everything is correct. There should be two signatures with different return types.

**Comment by ersimont** [comment]
Sorry I wasn't available for this discussion.  As thorn0 said, they key is that the return type is different when no parameter is passed in, so that's why they deserve separate definitions.

**Pull request merged**